cpp/lib/src/MiscUtil.cc

```diff
@@ -293,7 +293,7 @@ bool NormaliseDOI(const std::string &doi_candidate, std::string * const normalis
 }
 
 
-const ThreadSafeRegexMatcher CONTAINS_DOI_MATCHER("(10[.][0-9]{4,}(?:[.][0-9]+)*/(?:(?![\"&\\'<>])\\S)+)");
+const ThreadSafeRegexMatcher CONTAINS_DOI_MATCHER("(10.\\d{4,9}/[-._;()/:A-Z0-9]+)");
```

Comment:
We only use all caps for scalar constants.

Reply:
c.f. definition of DOI matcher in the same file.